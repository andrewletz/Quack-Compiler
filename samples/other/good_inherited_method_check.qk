class C1() {
    def TellMeAboutYourself() : String {
        return """I am a C1\n""";
    }
}

class C2() extends C1 {
    def TellMeAboutYourself() : String {
        return """I am a C2. I inherit C1.\n""";
    }
}

class C3() extends C1 {
    def TellMeAboutYourself() : String {
        return """I am a C3. I inherit C1.\n""";
    }
}

class C4() extends C2 {
    def TellMeAboutYourself() : String {
        return """I am a C4. I inherit C2.\n""";
    }
}

x : C1 = C4();
x.TellMeAboutYourself().PRINT();

x = C2();
x.TellMeAboutYourself().PRINT();

x = C3();
x.TellMeAboutYourself().PRINT();

x = C1();
x.TellMeAboutYourself().PRINT();
