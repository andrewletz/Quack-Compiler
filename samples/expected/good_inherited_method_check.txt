I am a C4. I inherit C2.
I am a C2. I inherit C1.
I am a C3. I inherit C1.
I am a C1
